### PR TITLE
KTL-1241 chore: added eslint&prettier, run them on pre-commit

### PR DESCRIPTION
**Pull request by berezinant**
The idea is to
- add standard eslint & prettier in order to increase code quality,
- and be able to catch defects like the usage of css module class that does not exist in styles.

The second is achieved by generating `*.css.d.ts` files for css modules and running typescript compiler on pre-commit.
It works like that:

why

### CODE REVIEW

**Comment by berezinant** [comment]
Here is a small demo showing no regress in markup (at least at the first glance) https://monosnap.com/file/Q9bpUPGFYROUtlkty3RFXJyCLo7SrZ

File: src/components/Link/index.tsx [hunk 1]
```diff
@@ -1,64 +1,63 @@
...
+  const commonProps = {
+    ...props,
+    className: cn(className, linkClassName),
+  };
+
+  if (isExternal || href.startsWith('mailto:')) {
+    return <a target="_blank" rel="noopener noreferrer" {...commonProps} />;
+  }
+
+  return <GatsbyLink to={href} {...commonProps} />;
```
Comment (berezinant) [inline src/components/Link/index.tsx:56]: Probably this is the only nontrivial change in code due to typescript checks

Comment (nikpachoo) [reply src/components/Link/index.tsx:56]: ✅ Checked, no issue found.

**Review by nikpachoo — APPROVED** [review]
It's hard to review all the changes thoroughly due to the size of the PR, so I reviewed the configuration files and part of the CSS and TSX files. Look good for me.

**Pull request merged**